polkadot/node/core/pvf/src/testing.rs

```diff
@@ -62,6 +62,29 @@ pub fn validate_candidate(
 pub fn get_and_check_worker_paths() -> (PathBuf, PathBuf) {
 	// Only needs to be called once for the current process.
 	static WORKER_PATHS: OnceLock<Mutex<(PathBuf, PathBuf)>> = OnceLock::new();
+
+	fn maybe_build_workers(build_prep: bool, build_exec: bool) {
+		let mut build_args = vec!["build", "--package=polkadot"];
+		if build_prep {
+			build_args.push("--bin=polkadot-prepare-worker");
+		}
+		if build_exec {
+			build_args.push("--bin=polkadot-execute-worker");
+		}
```

Comment:
Any reason not to simplify this and always build both workers, getting rid of the `build_*` arguments? I think it'd be rather rare for one to be up-to-date, but not the other. Plus, if something is already built, `cargo` will skip building it again. 

Maybe we could always try to build the workers. That would also satisfy my request [here](https://github.com/paritytech/polkadot-sdk/pull/2058#discussion_r1376050050) and it's simpler than doing a version check. The version check with the commit hash would only work if you've made a commit since the last build. But if you've changed code without committing, the version check would still pass without a rebuild being requested.

Reply:
Yeah I was also thinking about the same. However, then we should probably entirely drop any checking of versions or whatever and just build them.

Reply:
Let me test it out.